src/sdk/nsfs_glacier_backend/backend.js

```diff
@@ -153,7 +169,22 @@ class GlacierBackend {
      * @returns {nb.RestoreStatus | undefined}
      */
     static get_restore_status(xattr, now, file_path) {
-        if (xattr[GlacierBackend.STORAGE_CLASS_XATTR] !== s3_utils.STORAGE_CLASS_GLACIER) return;
+		if (GlacierBackend.storage_class_from_xattr(xattr, config.NSFS_GLACIER_USE_DMAPI) !== s3_utils.STORAGE_CLASS_GLACIER) {
+			return;
+		}
+
+        if (config.NSFS_GLACIER_USE_DMAPI) {
+            if (xattr[GlacierBackend.GPFS_DMAPI_XATTR_TAPE_PREMIG]) {
+                const expiry = new Date();
+                expiry.setDate(expiry.getDate() + config.S3_RESTORE_REQUEST_MAX_DAYS);
```

Comment:
I think we should have a specific config for this amount of days `config.NSFS_GLACIER_DMAPI_PMIG_DAYS` which will be initialized by default to `config.S3_RESTORE_REQUEST_MAX_DAYS` (which is 30), but could be overridden in config.json. The reason is that I'm not sure that the default we pick now would make sense to all cases. Perhaps in some cases replying with now+1day make better sense for example.

Also we can add comment in the code to explain:

```js
// we do not know for how long the file is going to remain available,
// the expiry is set to now + fixed config, which means it's always appears
// to the user with the same amount of time left before it expires.
expiry.setDate(expiry.getDate() + config.NSFS_GLACIER_DMAPI_PMIG_DAYS);
```